e2e/framework/testserver.go

```diff
@@ -417,11 +418,29 @@ func (t *TestServer) SendRawTx(
 	signer := crypto.NewEIP155Signer(100)
 	client := t.JSONRPC()
 
+	// if there is no from addr passed in we can find it in test server configuration
+	var blankAddr types.Address
+	if tx.From == blankAddr {
+		tx.From = t.Config.PremineAccts[0].Addr
+	}
+
 	nextNonce, err := client.Eth().GetNonce(web3.Address(tx.From), web3.Latest)
 	if err != nil {
 		return nil, err
 	}
 
+	if tx.GasPrice == big.NewInt(int64(0)) || tx.GasPrice == nil {
+		tx.GasPrice = big.NewInt(DefaultGasPrice)
+	}
+
+	if tx.Gas == 0 {
+		tx.Gas = DefaultGasLimit
+	}
+
+	if tx.Value == big.NewInt(0) || tx.Value == nil {
+		tx.Value = big.NewInt(0)
+	}
```

Comment:
Why the default values?

I don't think we should override values like this in a common func such as `SendRawTx`

Reply:
tx.Value needs to be set, as SignTx function is panicking when it tries to do tx.Copy and the tx.Value is not set. [here](https://github.com/0xPolygon/polygon-edge/blob/48ec5c46a89b5e57fc538eac9d8e57ebdb350286/types/transaction.go#L53)
 
As for tx.Gas and tx.GasPrice , these same checks are done in SendTxn function ( which we're using right now ) , and I've just implemented them here. Should we remove this ?

Reply:
Resolving because it's outdated